server/src/error.rs

```diff
@@ -0,0 +1,26 @@
+use core::{
+    error::Error,
+    fmt::{Debug, Display},
+};
+
+use crate::platform::{crypto::Drbg, TpmContextDeps};
+
+pub enum ServerError<Deps: TpmContextDeps> {
+    Drbg(<Deps::Drbg as Drbg>::Error),
+}
+
+impl<Deps: TpmContextDeps> Debug for ServerError<Deps> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Self::Drbg(arg0) => f.debug_tuple("Drbg").field(arg0).finish(),
+        }
+    }
+}
+
+impl<Deps: TpmContextDeps> Display for ServerError<Deps> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:#?}", self)
```

Comment:
Normally display is supposed to be a human-readable error, like "failed to reseed Drbg" or something like that. I'd say calling into `Debug` like this is kind of an anti-pattern.

I'd rather see `Debug` derived, and then `Display` used to print human-readable error messages.